massa-ledger-exports/src/key.rs

```diff
@@ -70,35 +89,34 @@ impl KeySerializer {
     }
 }
 
-impl Serializer<Vec<u8>> for KeySerializer {
+impl Serializer<Key> for KeySerializer {
     /// ```
     /// use massa_models::address::Address;
-    /// use massa_ledger_exports::{KeySerializer, DATASTORE_IDENT};
+    /// use massa_ledger_exports::{KeySerializer, KeyType, Key};
     /// use massa_serialization::Serializer;
     /// use massa_hash::Hash;
     /// use std::str::FromStr;
     ///
     /// let mut serialized = Vec::new();
     /// let address = Address::from_str("AU12dG5xP1RDEB5ocdHkymNVvvSJmUL9BgHwCksDowqmGWxfpm93x").unwrap();
     /// let store_key = Hash::compute_from(b"test");
-    /// let mut key = Vec::new();
-    /// key.extend(address.prefixed_bytes());
-    /// key.push(DATASTORE_IDENT);
-    /// key.extend(store_key.to_bytes());
+    /// let mut key = Key::new_datastore(&address, Some(store_key.into_bytes().to_vec()));
     /// KeySerializer::new().serialize(&key, &mut serialized).unwrap();
     /// ```
-    fn serialize(&self, value: &Vec<u8>, buffer: &mut Vec<u8>) -> Result<(), SerializeError> {
-        let limit = ADDRESS_SIZE_BYTES + 1;
-        buffer.extend(&value[..limit]);
-        if value[limit - 1] == DATASTORE_IDENT {
-            if value.len() > limit {
-                self.vec_u8_serializer
-                    .serialize(&value[limit..].to_vec(), buffer)?;
-            } else {
+    fn serialize(&self, value: &Key, buffer: &mut Vec<u8>) -> Result<(), SerializeError> {
+        buffer.extend(&value.address.prefixed_bytes());
```

Comment:
Any reason why AddressSerializer is not used here?

Reply:
Fixed with https://github.com/massalabs/massa/pull/3455/commits/2014ce50d37f6c73538ba67f52054767fad38589 (and https://github.com/massalabs/massa/pull/3455/commits/a7da4a7c68bdbe92ce65bb5ff2bd842ba87122c7)